### PR TITLE
Unite SectionCreationForm and Se…

### LINKED ISSUE #357: Unite SectionCreationForm and SectionUpdateForm
This is almost similar components

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes

-   [x] Refactor

## Related issues

Resolve #357

### CODE REVIEW

**Comment by Katrin-kudryash** [comment]
https://github.com/taskany-inc/hire/assets/81121562/031dfa06-2f88-4dc3-af65-1db89beb626c

File: src/components/CreateOrUpdateSectionForm.tsx/CreateOrUpdateSectionForm.tsx [hunk 1]
```diff
@@ -121,11 +148,22 @@ export function SectionCreationForm({ candidate, interviewId, sectionType }: Pro
...
         setSearch('');
     };
+
     const onSchedulableToggle = (arg: boolean) => {
         setSchedulable(arg);
         setSearch('');
     };
 
+    const onCancelButton = () => {
+        if (!section) {
```
[the remark this answers is not in the document]
Comment (nyet-ty) [reply src/components/CreateOrUpdateSectionForm.tsx/CreateOrUpdateSectionForm.tsx:158]: I don't think it's needed on SectionCreationForm

[the remark this answers is not in the document]
Comment (tsumo) [reply src/components/CreateOrUpdateSectionForm.tsx/CreateOrUpdateSectionForm.tsx:158]: You mean Cancel Button?

[the remark this answers is not in the document]
Comment (nyet-ty) [reply src/components/CreateOrUpdateSectionForm.tsx/CreateOrUpdateSectionForm.tsx:158]: yes

[the remark this answers is not in the document]
Comment (Katrin-kudryash) [reply src/components/CreateOrUpdateSectionForm.tsx/CreateOrUpdateSectionForm.tsx]: I fixed. What to do?

File: src/components/CreateOrUpdateSectionForm.tsx/CreateOrUpdateSectionForm.tsx [hunk 2]
```diff
@@ -88,30 +109,49 @@ export function SectionCreationForm({ candidate, interviewId, sectionType }: Pro
...
+    const updateSection: SubmitHandler<CreateOrUpdateSection> = useCallback(
+        async (data) => {
+            if (!section) {
+                return;
+            }
+            await sectionUpdateMutation.mutateAsync({ data });
+
+            await router.push(pageHrefs.interviewSectionView(interviewId, section.id));
+        },
+        [interviewId, router, schedulable, sectionUpdateMutation, sectionType.id],
```
Comment (nyet-ty) [inline src/components/CreateOrUpdateSectionForm.tsx/CreateOrUpdateSectionForm.tsx:121]: GitHubActions / build tells you to add section to dependencies of useCallback

**Review by tsumo — APPROVED** [review]

**Review by nyet-ty — CHANGES_REQUESTED** [review]

**Review by nyet-ty — APPROVED** [review]

**Pull request merged**